src/requestforattestation/RequestForAttestation.ts

```diff
@@ -61,6 +63,95 @@ function getHashRoot(leaves: Uint8Array[]): Uint8Array {
   return hash(result)
 }
 
+function compressNonceAndHash(nonceHash: any[]): any {
+  if (nonceHash.length === 0) {
+    return {
+      hash: nonceHash[0],
+    }
+  }
+  return {
+    nonce: nonceHash[0],
+    hash: nonceHash[1],
+  }
+}
+
+export function compressClaimHashTree(
+  rfa: IRequestForAttestation
+): IRequestForAttestation['claimHashTree'] {
+  const result = {}
+  const claimTree = rfa.claimHashTree
+
+  Object.keys(claimTree).forEach(entryKey => {
+    result[entryKey] = Object.values(claimTree[entryKey])
+  })
+  return result
+}
+
+export function decompressClaimHashTree(
+  rfa: IRequestForAttestation[]
+): IRequestForAttestation['claimHashTree'] {
+  const result = {}
+
+  Object.keys(rfa).forEach(entryKey => {
+    result[entryKey] = compressNonceAndHash(Object.values(rfa[entryKey]))
+  })
+  return result
+}
+
+export function compressClaimContents(rfa: IRequestForAttestation): any {
+  // should go into the claim module.
+  return Object.values(rfa.claim)
+}
+
+export function decompressClaimContents(rfa: any): any {
+  // should go into the claim module.
+  return {
+    cTypeHash: rfa[0],
+    contents: rfa[1],
+    owner: rfa[2],
+  }
+}
+
+function compressLegitimaition(leg: AttestedClaim): any {
+  return leg.compress()
+}
+
+export function decompressLegitimaition(leg: any): any {
+  if (!leg[0]) {
+    return []
+  }
+  return [decompressAttestedClaim(leg[0])] // need to test for multiple leg
+}
+
+export function compressRequestForAttestation(
+  reqForAttest: IRequestForAttestation
+): any {
+  return [
+    compressClaimContents(reqForAttest),
```

Comment:
Question (please see w/ @tjwelde):
Do we want to do data validation here? And in general for this compression/decompression feature?

Reply:
Validation is covered by `fromRequest` and similar methods that call the constructor that in turns performs validation.